File: 112/InsertionAtBeginning.py
class Node:
    def __init__(self,n:int):
        self.next = None
        self.val = n

def main():
    n = int(input())
    head = None
    for i in range(n):
        k = int(input())
        head = insertAtBeginning(head,k)

    iter = head
    while(iter!=None):
        print(iter.val,end=" ")
        iter=iter.next


def insertAtBeginning(head: Node, n: int) -> Node:
    newNode = Node(n)
    newNode.next = head
    return newNode

if __name__ == "__main__":
    main()

File: 112/InsertionAtBeginning.dart
import 'dart:io';

class Node{
  int val;
  Node? next;
  Node(this.val);

}

void main(){
  int n = int.parse(stdin.readLineSync()!);
  Node head;
  for(int i=0;i<n;i++){
    int k=int.parse(stdin.readLineSync()!);
    head = insertAtBeginning(head,k);
  }
}

Node insertAtBeginning(Node head, int val){
  Node newNode = Node(val);
  newNode.next = head;
  return newNode;
}